src/Raven.Server/Documents/Subscriptions/SubscriptionConnectionBase.cs

```diff
@@ -0,0 +1,490 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Subscriptions;
+using Raven.Client.Exceptions.Commercial;
+using Raven.Client.Exceptions.Database;
+using Raven.Client.Exceptions.Documents.Subscriptions;
+using Raven.Client.Http;
+using Raven.Client.ServerWide.Commands;
+using Raven.Server.Documents.TcpHandlers;
+using Raven.Server.Json;
+using Raven.Server.Rachis;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Logging;
+using Sparrow.Utils;
+
+namespace Raven.Server.Documents.Subscriptions
+{
+    public abstract class SubscriptionConnectionBase : IDisposable
+    {
+        private static readonly byte[] Heartbeat = Encoding.UTF8.GetBytes("\r\n");
+        public const long NonExistentBatch = -1;
+
+        protected readonly ServerStore _serverStore;
+        private readonly IDisposable _tcpConnectionDisposable;
+        internal readonly (IDisposable ReleaseBuffer, JsonOperationContext.MemoryBuffer Buffer) _copiedBuffer;
+
+        internal SubscriptionWorkerOptions _options;
+        internal readonly Logger _logger;
+
+        public readonly ConcurrentQueue<string> RecentSubscriptionStatuses = new ConcurrentQueue<string>();
+        internal bool _isDisposed;
+        public SubscriptionWorkerOptions Options => _options;
+        public SubscriptionException ConnectionException;
+        public long SubscriptionId { get; set; }
+        public string Database;
+        public ShardData Shard;
```

Comment:
this is wrong, base/abstract class should be 'neutral'

Reply:
done